lib/internal/process/per_thread.js

```diff
@@ -423,11 +428,18 @@ function toggleTraceCategoryState(asyncHooksEnabled) {
   }
 }
 
+const { arch, platform, version } = process;
+const language = Intl?.Collator().resolvedOptions().locale || 'en-US';
```

Comment:
I think the language cannot be cached because ICU data can be dynamically loaded at runtime. Although, I am not sure if this is the right implementation in the first place, not sure if V8's Intl would pick up LC environment variables...

Reply:
hmm, interesting - since i believe the language for a webpage is set, and changing it reloads the page, Navigator shouldn't be able to report a different language within the life of a program.

I can't speak to whether the pre-existing implementation is correct, but hopefully if so that can be fixed in a separate PR.

Reply:
This is cached at build time, so it would the same as the environment in the build machine.

Reply:
ah, well then that's definitely not correct.

what do you suggest for caching the language at app startup, before user code is run?

Reply:
Updated!